src/wallet-watcher.ts

```diff
@@ -64,219 +89,112 @@ export const determineWalletAddress = (wallet: Wallet) => {
-
-  return walletBalances.reduce((acc: WalletStatus[], settledPromise: PromiseSettledResult<WalletStatus>) => {
-    if (settledPromise.status === 'rejected') {
-      logging.log(`Failed to get wallet balance.`, 'ERROR', `Error: ${settledPromise.reason.message}.`);
-      return acc;
-    }
-
-    return [...acc, settledPromise.value];
-  }, []);
-};
-
-/**
- * Takes a list of wallets with their embedded balances, compares the balance to configured minimums and executes top
- * up transactions if low balance criteria are met.
- *
- * @param wallet a wallet object containing a wallet address and balance
- * @param config parsed config.json
- * @param globalSponsors a set of wallets used as the source of funds for top-ups. Different wallets represent funds on different chains.
- */
-export const checkAndFundWallet = async (wallet: WalletStatus, config: Config, globalSponsors: GlobalSponsor[]) => {
-  try {
-    const globalSponsor = globalSponsors.find((sponsor) => sponsor.chainId === wallet.chainId);
-
-    // Close previous cycle alerts
-    await opsGenie.closeOpsGenieAlertWithAlias(
-      `freshly-topped-up-${wallet.address}-${wallet.chainName}`,
-      config.opsGenieConfig
-    );
-
-    if (!globalSponsor) {
-      await opsGenie.sendToOpsGenieLowLevel(
-        {
-          message: `Can't find a valid global sponsor for ${wallet.address} on ${wallet.chainName}`,
-          alias: `no-sponsor-${wallet.address}-${wallet.chainName}`,
-          priority: 'P1',
-        },
-        config.opsGenieConfig
-      );
-      return;
-    }
-
-    await opsGenie.closeOpsGenieAlertWithAlias(
-      `no-sponsor-${wallet.address}-${wallet.chainName}`,
-      config.opsGenieConfig
-    );
-
-    const walletBalanceThreshold = parseEther(globalSponsor.lowBalance);
-    if (wallet.balance.gt(walletBalanceThreshold)) {
-      return;
-    }
-
-    const globalSponsorBalanceThreshold = parseEther(globalSponsor.globalSponsorLowBalanceWarn);
-    if ((await globalSponsor.sponsor.getBalance()).lt(globalSponsorBalanceThreshold)) {
-      await opsGenie.sendToOpsGenieLowLevel(
-        {
-          message: `Low balance on primary top-up sponsor for chain ${wallet.chainName}`,
-          alias: `low-master-sponsor-balance-${wallet.chainName}`,
-          priority: 'P3',
-        },
-        config.opsGenieConfig
-      );
-    } else {
-      await opsGenie.closeOpsGenieAlertWithAlias(
-        `low-master-sponsor-balance-${wallet.chainName}`,
-        config.opsGenieConfig
-      );
-    }
-
-    const [logs, gasTarget] = await getGasPrice(wallet.provider, config.chains[wallet.chainId].options);
-    logs.forEach((log) =>
-      logging.log(log.message, log.level === 'INFO' || log.level === 'ERROR' ? log.level : undefined)
-    );
+  // Unique wallets with highest lowThreshold by address and chainId
+  const uniqueWalletsPerChain = uniqBy(
+    [...walletsWithDerivedAddresses].sort((a, b) => b.lowThreshold.value - a.lowThreshold.value),
```

Comment:
🕵️ 
```suggestion
    walletsWithDerivedAddresses.sort((a, b) => b.lowThreshold.value - a.lowThreshold.value),
```

Reply:
(I was hoping this would slide in unnoticed 🤭 ) 

The Array.sort() method mutates the original array therefore I was getting an es-lint warning. Spreading the elements of the array into a new array was my hacky attempt to create a copy of the array before sorting it.

Reply:
> (I was hoping this would slide in unnoticed

😁 

> The Array.sort() method mutates the original array therefore I was getting an es-lint warning. Spreading the elements of the array into a new array was my hacky attempt to create a copy of the array before sorting it.

Ah true, I had a feeling you might have had a reason to leave it like that